runtime/sema/checker.go

```diff
@@ -546,6 +547,9 @@ func (checker *Checker) checkTransfer(transfer *ast.Transfer, valueType Type) {
 	}
 }
 
+// This method is only used for checking invocation-expression.
+// This is also temporary, until the type inferring support is added for func arguments
+// TODO: Remove this method
 func (checker *Checker) checkTypeCompatibility(expression ast.Expression, valueType Type, targetType Type) bool {
```

Comment:
This function is no longer used, except in function-invocation expression checking.

The functionality inside this method is now distributed to the respective expressions.
e.g:  
the logic inside the `case *ast.IntegerExpression:` of the switch-case in this method, is now moved to `func (checker *Checker) VisitIntegerExpression()` method.